pkg/controller/user-cluster-controller-manager/resources/controller.go

```diff
@@ -106,6 +106,7 @@ func Add(
 		versions:          versions,
 		caBundle:          caBundle,
 		userClusterMLA:    userClusterMLA,
+		cloudProvider:     types2.CloudProvider(cloudProviderName),
```

Comment:
What is the reason for using a machine-controller type/function here? How is this going to behave with a provider that's supported by KKP, but not MC, for example, BringYourOwnCloud?

Reply:
The reason behind it is that we already had all the needed providers in that pkg. Btw, having BringYourOwn and casting it to CloudProvider is fine, the string is simply cast to `CloudProvider` type. This cast was needed to compare the current CloudProvider with the ones defined in the mc pkg. If you think this is a kind of unclean approach, I can re-declare those constants in a pkg in KKP and use them.